persist/fs/seek.go

```diff
@@ -368,25 +362,58 @@ func (s *seeker) Seek(id ts.ID) (checked.Bytes, error) {
 		n += remainder
 	}
 
-	if n != int(entry.size) {
+	if n != int(entry.Size) {
 		return nil, errReadNotExpectedSize
 	}
 
 	// NB(r): _must_ check the checksum against known checksum as the data
 	// file might not have been verified if we haven't read through the file yet.
-	if entry.checksum != digest.Checksum(data.Get()) {
+	if entry.Checksum != digest.Checksum(data.Get()) {
 		return nil, errSeekChecksumMismatch
 	}
 
 	return data, nil
 }
 
-func (s *seeker) SeekOffset(id ts.ID) int {
-	entry, exists := s.indexMap[id.Hash()]
-	if !exists {
-		return -1
+func (s *seeker) SeekIndexEntry(id ts.ID) (IndexEntry, error) {
+	offset, err := s.indexLookup.getNearestIndexFileOffset(id)
+	// Should never happen, either something is really wrong with the code or
+	// the file on disk was corrupted
+	if err != nil {
+		return IndexEntry{}, err
 	}
-	return int(entry.offset)
+
+	stream := msgpack.NewDecoderStream(s.indexMmap[offset:])
+	s.decoder.Reset(stream)
+
+	// Prevent panic's when we're scanning to the end of the buffer
+	for stream.Remaining() != 0 {
+		entry, err := s.decoder.DecodeIndexEntry()
+		// Should never happen, either something is really wrong with the code or
+		// the file on disk was corrupted
+		if err != nil {
+			return IndexEntry{}, err
+		}
+		comparison := bytes.Compare(entry.ID, id.Data().Get())
```

Comment:
Can we move the `id.Data().Get()` call to before this for loop and keep the ref in a local var?  Doing ref counting for each lookup might be expensive.

Reply:
good call